src/app/core/token-interceptor.service.ts

```diff
@@ -0,0 +1,24 @@
+import {Inject, Injectable} from '@angular/core';
+import {HttpEvent, HttpHandler, HttpRequest} from '@angular/common/http';
+import {Observable} from 'rxjs';
+import {AuthConfig, AuthTokenConfig} from '@core';
+
+@Injectable()
+export class TokenInterceptorService {
+
+  constructor(@Inject(AuthConfig) private config: AuthTokenConfig) {
+  }
+
+  intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+    const token = localStorage.getItem('app_token');
+    if (!request.url.startsWith(this.config.ApiUrl)
+      || (request.url.startsWith(this.config.ApiUrl + '/public'))
+      || !token) {
+      return next.handle(request);
+    }
+    const clonedRequest = request.clone(
+      {headers: request.headers.set('x-auth-token', JSON.parse(token)['app_token'])});
```

Comment:
Process `JSON.parse(token)['app_token']` outside of here so we can throw an error in the token format is wrong.